### ISSUE TITLE
project question 

### ISSUE THREAD

**Issue by hereisjulia** [hereisjulia/R_Visualization#1]
@tpemartin

https://github.com/hereisjulia/R_Visualization/blob/7f91addc1016c83fe707b80287d3b0d098183183/Tasks/Energy.Rmd#L14

**Comment by hereisjulia** [comment]
@tpemartin 
老師好！
我是資料視覺化處理課程的蔡亞樵，
不好意思，實體周一課程我無法出席，想詢問是否可以固定在其他時間拜訪老師？(雙周一次)
另外，在課程部分多沒有問題，
但我發現當我看到一個新的data的時候就會整個傻掉不知道要如何處理，
我可以多準備幾個data在下周問老師嗎？

https://github.com/hereisjulia/R_Visualization/blob/85557bc192912f146f17befda9ec756a094b70d8/Tasks/world%20GHG%20emissions/GHG_emissions.Rmd#L11
比如這個，我就會不確定要如何下手設定x,y，想要用dplyr調整但抓不太到方向~

**Comment by tpemartin** [comment]
@hereisjulia please check

https://github.com/tpemartin/R_Visualization/blob/main/GHG_emissionsMartin.Rmd

I will send out the consultation time frame later today. Thanks for the reminder.

**Comment by hereisjulia** [comment]
@tpemartin 
謝謝老師~!!這裡有個好奇的問題~

https://github.com/tpemartin/R_Visualization/blob/ff8455e2d215e235f91d43d99df50379b8d937b3/GHG_emissionsMartin.Rmd#L117
>記得之前在課堂上有說在資料視覺化時多會希望用long form以求可以使用group，
但現實中像我這個案例是否就算是：如果不將資料格式更改為long form就無法使用? 
所以一般在進行資料視覺化的資料清理時，第一步就是將資料型態更正為long form嗎~?

超感謝老師~

**Comment by tpemartin** [comment]
Yes. You encountered exactly the case I mentioned in the class. Good job.

**Comment by hereisjulia** [comment]
<img width="501" alt="2022-05-10 (7@tpemartin " src="https://user-images.githubusercontent.com/47810433/167628460-d67818d1-5b18-4273-a42a-e4d4910c428b.png">
@tpemartin

**Comment by hereisjulia** [comment]
@tpemartin 
2022/5/24 Online inquiry
https://github.com/hereisjulia/R_Visualization/blob/692b2b94a482b0c0092ff77630d59319337dd908/Tasks/Final%20Project/FinalProject.Rmd#L32

**Comment by hereisjulia** [comment]
@tpemartin 
在做圖時遇到一些疑惑想詢問老師
不好意思打擾老師了~!

這邊是畫圖部分的code: 
https://github.com/hereisjulia/R_Visualization/blob/eead1c5daa11d91001293857e2d60e39a60f0d84/Tasks/Final%20Project/FinalProject.Rmd#L141-L221

圖的內容是雙軸圖，資料有3個，分別是：
1. 台灣總能源消耗量 > geom_area (左軸)
2. 台灣進口能源類別 > geom_area (左軸)
3. 進口能源占比(%)   > geom_line (右軸)

但我發現原本可以完整呈現的資料2，在把右軸加入之後，會有一些不見 (五個種類的資料不會全部出現)
https://github.com/hereisjulia/R_Visualization/blob/eead1c5daa11d91001293857e2d60e39a60f0d84/Tasks/Final%20Project/FinalProject.Rmd#L216-L221

**Comment by tpemartin** [comment]
Hi I'm in the road now. Will respond to you on Sunday. Sorry for the inconvenience.

**Comment by hereisjulia** [comment]
Of course! Take your time, please.
Thanks a lot!

**Comment by tpemartin** [comment]
我看了一下你的程式，首先，geom_area() 用了好幾遍，你可能要考慮把資料dplyr::bind_rows(df1, df2, ...dfx) 在一起再用一個geom_area去繪，這樣應該不會有消失的問題。

The example code is:
```
library(dplyr)
finalProj$BureauEnergy$data$Dom_Energy_Consumption |>
  select(month, Consump_kloe) |>
  rename(
    x=month, y=Consump_kloe
  ) |>
  mutate(
    source="Dom_Energy_Consumption"
  ) -> .df1

finalProj$BureauEnergy$data$Imp_EnergySource |>
  select(month, Import_kloe) |>
  rename(
    x=month, y=Import_kloe
  ) |>
  mutate(
    source="Imp_EnergySource"
  ) -> .df2

## to use bind_rows, columns with the same name will be stacked on top of each other. that's why we need to rename columns across data frames earlier.
dplyr::bind_rows(
  .df1, .df2
) -> .dfall
View(.dfall)

ggplot(data=.dfall)+
  geom_area(
    aes(x=x, y=y, fill=source)
  ) +
  scale_fill_manual(
    limits=c("Dom_Energy_Consumption", "Imp_EnergySource"),
    values=c("pink", "green")
  )

```

**Comment by hereisjulia** [comment]
@tpemartin 謝謝老師~
我試過後發現是資料單位超過我設定的軸單位所以沒有顯示~!!
